File: Data Structures & Algorithms/max-water-container/submission-6.java
class Solution {
    public int maxArea(int[] heights) {
        int left = 0;
        int right = heights.length - 1;
        int maxArea = 0;

        while(left < right) {
            int minHeight = Math.min(heights[left], heights[right]);
            maxArea = Math.max(maxArea, (right-left) * minHeight);
            System.out.println(heights[left] + " " + heights[right]);
            if (heights[left] < heights[right]) {
                left++;
            } else {
                right--;
            }
        }
        return maxArea;
    }
}

File: Data Structures & Algorithms/binary-tree-maximum-path-sum/submission-6.py
class Solution:
    
    def maxPathSum(self, root: Optional[TreeNode]) -> int:
        
        maxPath=[float("-inf")]

        def maxSum(root):
            if root is None:
                return 0

            leftPath = maxSum(root.left)
            rightPath = maxSum(root.right)

            maxLeftPath = max(leftPath, 0)
            maxRightPath = max(rightPath, 0)

            maxPath[0] = max(maxPath[0], root.val + maxLeftPath + maxRightPath)
            
            return max(root.val + maxLeftPath, root.val + maxRightPath)
        
        maxSum(root)
        return maxPath[0]